### ISSUE TITLE
Custom cells do not render with virtualization enabled

### ISSUE THREAD

**Issue by kaiqueaguilar** [KevinVandy/material-react-table#568]
### material-react-table version

1.13.0

### react & react-dom versions

18.2.0

### Without row virtualization:
<img width="971" alt="image" src="https://github.com/KevinVandy/material-react-table/assets/92055543/035595e3-d69c-4df0-9856-99a9c18f46ba">

### With row virtualization:

<img width="973" alt="image" src="https://github.com/KevinVandy/material-react-table/assets/92055543/a7ad3cce-0028-4c74-9407-2811eed7ffe7">

### Column:
```
{
  accessorKey: "valorPercentual",
  id: "barValorPercentual",
  header: "Participação",
  size: 200,
  Cell: ({ cell }) => (<LinearBar variant="determinate" value={10} />),
}
```

### Minimal, Reproducible Example - (Optional, but Recommended)

```
import { LinearProgress, linearProgressClasses, styled } from "@mui/material";
import { MRT_ColumnDef, MaterialReactTable as Table } from "material-react-table";

export function TableRankingIndicadores(): JSX.Element {
const data = [
    {
        "id": "1",
        "agrupamentoDescricao": "test 1",
        "valorIndicador": 6463576.57,
        "valorPercentual": 40.39,
        "referenciaTipo": "FILIAL",
        "indicadorTipo": "VENDALIQUIDA"
    },
    {
        "id": "2",
        "agrupamentoDescricao": "test 2",
        "valorIndicador": 4349749.56,
        "valorPercentual": 27.18,
        "referenciaTipo": "FILIAL",
        "indicadorTipo": "VENDALIQUIDA"
    },
    {
        "id": "3",
        "agrupamentoDescricao": "test 3",
        "valorIndicador": 3536546.60,
        "valorPercentual": 22.10,
        "referenciaTipo": "FILIAL",
        "indicadorTipo": "VENDALIQUIDA"
    },
    {
        "id": "4",
        "agrupamentoDescricao": "test 4",
        "valorIndicador": 1651639.73,
        "valorPercentual": 10.32,
        "referenciaTipo": "FILIAL",
        "indicadorTipo": "VENDALIQUIDA"
    },
    {
        "id": "5",
        "agrupamentoDescricao": "test 5",
        "valorIndicador": 0,
        "valorPercentual": 0,
        "referenciaTipo": "Filial",
        "indicadorTipo": "VendaLiquida"
    }
]

const LinearBar = styled(LinearProgress)(({ theme }) => ({
  height: 15,
  borderRadius: 4,
  [`&.${linearProgressClasses.colorPrimary}`]: {
    backgroundColor: theme.palette.divider,
  },
  [`& .${linearProgressClasses.bar}`]: {
    borderRadius: 4,
    backgroundColor: theme.palette.primary.main,
  },
}));

  const columns: MRT_ColumnDef[] = [
    {
      accessorKey: "agrupamentoDescricao",
      id: "agrupamentoDescricao",
      header: "Filial",
      size: 300,
    },
    {
      accessorKey: "valorIndicador",
      id: "valorIndicador",
      header: "Valor",
      size: 200,
    },
    {
      accessorKey: "valorPercentual",
      id: "valorPercentual",
      header: "Percentual",
      size: 100,
    },
    {
      accessorKey: "valorPercentual",
      id: "barValorPercentual",
      header: "Participação",
      size: 200,
      Cell: ({ cell }) => (<LinearBar variant="determinate" value={Number(cell.getValue())} />),
    },
  ];

  return (
    <Table
      data={data}
      columns={columns}
      enableStickyHeader
      enableStickyFooter
      enableBottomToolbar={false}
      enableColumnResizing
      enableColumnOrdering
      // enableRowVirtualization
      enableColumnDragging={false}
      enablePagination={false}
    />
  );
}
```

### Do you intend to try to help solve this bug with your own PR?

No, because I do not have time to dig into it

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by KevinVandy** [comment]
Maybe it just has to do with your component? It seems to work fine for me here: https://codesandbox.io/p/sandbox/github/KevinVandy/material-react-table/tree/main/apps/material-react-table-docs/examples/virtualized/sandbox

**Comment by jlundan** [comment]
I actually ran into this myself earlier, and I think it is just the `LinearProgress` component not being able to deal with flex positioning and getting zero width. I did a quick test implementation in my current app by throwing in sliders with row virtualization in one of my number-valued columns just to display the issue and attached few screenshots. Basically what I did to solve the issue, was to add set the width of MuiLinearProgress-root class to 100%:

```
.MuiLinearProgress-root {
  width: 100%;
}
```
I am not sure about the performance impact of this, but at least for my case (67 rows in my app, but when I encountered this originally, I had, I think, few hundred rows in the table) the performance seemed fine enough.

![Screenshot from 2023-06-17 03-22-45](https://github.com/KevinVandy/material-react-table/assets/1215642/745d0953-ea71-4dba-a4d1-60cbf46af0fe)
![Screenshot from 2023-06-17 03-14-44](https://github.com/KevinVandy/material-react-table/assets/1215642/3a2b0e68-8f31-4b21-bf87-9d0dcabb9abf)
